src/main/java/uk/gov/pay/directdebit/common/model/DirectDebitStateWithDetails.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.pay.directdebit.common.model;
+
+import uk.gov.pay.directdebit.payments.model.DirectDebitState;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class DirectDebitStateWithDetails<T extends DirectDebitState> {
```

Comment:
Not sure there is much benefit in having the `DirectDebit` prefix since we're in the DirectDebit project and everything is DirectDebit...

Reply:
I was following on from `DirectDebitState` but I take your point